### PR TITLE
Add system call filter bootstrap check

### PR DESCRIPTION
**Pull request by jasontedor**
Today if system call filters fail to install on startup, we log a
message but otherwise march on. This might leave users without system
call filters installed not knowing that they have implicitly accepted
the additional risk. We should not be lenient like this, instead clearly
informing the user that they have to either fix their configuration or
accept the risk of not having system call filters installed. This commit
adds a bootstrap check that if system call filters are enabled, they
must successfully install.

### CODE REVIEW

**Comment by jasontedor** [comment]
I will add a note to the 5.2.0 migration docs when I backport this.

File: core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java [hunk 1]
```diff
@@ -470,16 +471,44 @@ public String errorMessage() {
 
     }
 
-    abstract static class MightForkCheck implements BootstrapCheck.Check {
+    abstract static class IsSeccompInstalledCheck implements BootstrapCheck.Check {
+
+        // visible for testing
+        boolean isSeccompInstalled() {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java]: Is inheritance really needed for one method?

**Review by rjernst — APPROVED** [review]
Looks good.

File: core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java]: Definitely not, I started with this check and the might fork checks sharing more code but decided that I didn't like it and this is a remnant of that effort. I will remove.

Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java]: I pushed 9c84a7694b0a68e6972e3aac9b30686db4170350.

**Comment by jasontedor** [comment]
Thanks @rjernst.

**Pull request merged**